### PR TITLE
Add tag and value attributes to UncaughtThrowError

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: src/main/ruby/core/exception.rb [hunk 1]
```diff
@@ -166,6 +166,19 @@ def to_s
 end
 
 class UncaughtThrowError < ArgumentError
+  attr_reader :tag
+  attr_reader :value
+
+  def initialize(*args)
+    Rubinius::Type.check_arity(args.size, 2, -1)
```
Comment (eregon) [inline src/main/ruby/core/exception.rb]: I don't like manual arity checks as you know :)
Can it be expressed by with the method arguments/signature?

Comment (bjfish) [reply src/main/ruby/core/exception.rb]: @eregon Fixed, I've updated the method signature.

File: src/main/ruby/core/exception.rb [hunk 2]
```diff
@@ -166,6 +166,18 @@ def to_s
...
+  attr_reader :value
+
+  def initialize(tag, value, *rest)
+    @tag = tag
+    @value = value
+    super(*rest)
+  end
+
+  def to_s
+    sprintf(self.message, @tag)
```
Comment (eregon) [inline src/main/ruby/core/exception.rb]: Isn't it a problem that Exception#message calls `self.to_s` and here the reverse?

Comment (bjfish) [reply src/main/ruby/core/exception.rb]: My intention was to the the `message` field, fixed now.

**Pull request merged**